config/kernel-vfs-iov_iter.m4

```diff
@@ -201,4 +209,19 @@ AC_DEFUN([ZFS_AC_KERNEL_VFS_IOV_ITER], [
 		AC_DEFINE(HAVE_VFS_IOV_ITER, 1,
 		    [All required iov_iter interfaces are available])
 	])
+
+	dnl #
+	dnl # Kernel 6.5 introduces the iter_iov() function that returns the
+	dnl # __iov member of an iov_iter*. The iov member was renamed to this
+	dnl # __iov member, and is intended to be accessed via the helper
+	dnl # function now.
```

Comment:
It looks like the `iter_iov` helper was actually introduced in torvalds/linux@de4f5fed3f231 so it would be a 6.3 kernel API change.

In this case I think it would make sense to define our own version of `iter_iov()` as `#define iter_iov(iter) (iter)->iov` in `include/os/linux/spl/sys/uio.h` for older kernels.  Then we can use `iter_iov()` in the `HAVE_IOV_ITER_TYPE` case or elsewhere as needed.

Reply:
I agree with @behlendorf suggestion here. So if you look at top of `include/os/linux/spl/sys/uio.h` you will find we tend to do this quite a bit. So maybe adding something like `zfs_uio_iter_iov()` would be more useful here? Then if we need to in the future, we can use that to access the `struct iovec` from a `struct iov_iter`. We currently only do so from the `init` code, but there might come a time where it maybe useful in other places.

Reply:
Ok, I refactored this in a new commit, and also added a separate commit to perform a similar change for `iov_iter_type()`, which cleans up the `zpl_uio_init` code.